cmd/thanos/bucket.go

```diff
@@ -4,29 +4,37 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
-	"io/ioutil"
 	"os"
-	"path"
-	"path/filepath"
 	"text/template"
 	"time"
 
 	"cloud.google.com/go/storage"
 	"github.com/go-kit/kit/log"
-	"github.com/go-kit/kit/log/level"
 	"github.com/improbable-eng/thanos/pkg/block"
-	"github.com/improbable-eng/thanos/pkg/compact"
 	"github.com/improbable-eng/thanos/pkg/objstore"
 	"github.com/improbable-eng/thanos/pkg/objstore/gcs"
 	"github.com/improbable-eng/thanos/pkg/objstore/s3"
+	"github.com/improbable-eng/thanos/pkg/verifier"
 	"github.com/oklog/run"
 	"github.com/oklog/ulid"
-	"github.com/opentracing/opentracing-go"
+	opentracing "github.com/opentracing/opentracing-go"
 	"github.com/pkg/errors"
 	"github.com/prometheus/client_golang/prometheus"
 	"gopkg.in/alecthomas/kingpin.v2"
 )
 
+var (
+	allIssues = []string{
+		verifier.IndexIssueID,
+		verifier.OverlappedBlocksIssueID,
+	}
+
+	issuesMap = map[string]verifier.Issue{
+		verifier.IndexIssueID:            verifier.IndexIssue(),
+		verifier.OverlappedBlocksIssueID: verifier.OverlappedBlocksIssue(),
```

Comment:
Wait, why are we calling those functions anyway? They both don't do anything but return a function directly, so could just remove that level of indirection and write:

```
	issuesMap = map[string]verifier.Issue{
		verifier.IndexIssueID:            verifier.IndexIssue,
		verifier.OverlappedBlocksIssueID: verifier.OverlappedBlocksIssue,
	}
```